catalog/urls.py

```diff
@@ -1,8 +1,12 @@
 from django.urls import path
-from catalog.views import home, contacts
+from catalog.views import home, contacts, product_list
 from catalog.apps import CatalogConfig
 
 
 app_name = CatalogConfig.name
 
-urlpatterns = [path("", home, name="home"), path("contacts", contacts, name="contacts")]
+urlpatterns = [
+    path("", home, name="home"),
+    path("contacts", contacts, name="contacts"),
+    path('product_list/<int:id_product>', product_list, name='product_list')
```

Comment:
В конце нужно добавлять слеш, в ручке контактов тоже
```suggestion
    path('products/<int:product_id>/', product_details, name='product_details')
```